docs/core/metrics-v2.md

```diff
@@ -0,0 +1,889 @@
+* Add `FunctionName` and `Service` dimensions
+
+This has the advantage of keeping cold start metric separate from your application metrics, where you might have unrelated dimensions.
+
+## Advanced
+
+### Adding metadata
+
+You can add high-cardinality data as part of your Metrics log with `AddMetadata` method. This is useful when you want to search highly contextual information along with your metrics in your logs.
+
+!!! info
+    **This will not be available during metrics visualization** - Use **dimensions** for this purpose
+
+!!! info
+    Adding metadata with a key that is the same as an existing metric will be ignored
+
+=== "Function.cs"
+
+    ```csharp hl_lines="9"
+    using AWS.Lambda.Powertools.Metrics;
+    
+    public class Function {
+      
+      [Metrics(Namespace = ExampleApplication, Service = "Booking")]
+      public async Task<APIGatewayProxyResponse> FunctionHandler(APIGatewayProxyRequest apigProxyEvent, ILambdaContext context)
+      {
+        Metrics.AddMetric("SuccessfulBooking", 1, MetricUnit.Count);
+        Metrics.AddMetadata("BookingId", "683EEB2D-B2F3-4075-96EE-788E6E2EED45");
+        ...
+    ```
+
+=== "Example CloudWatch Logs excerpt"
+
+    ```json hl_lines="23"
+    {
+      "SuccessfulBooking": 1.0,
+      "_aws": {
+      "Timestamp": 1592234975665,
+      "CloudWatchMetrics": [
+        {
+      "Namespace": "ExampleApplication",
+      "Dimensions": [
+        [
+        "service"
+        ]
+      ],
+      "Metrics": [
+        {
+        "Name": "SuccessfulBooking",
+        "Unit": "Count"
+        }
+      ]
+        }
+      ]
+      },
+      "Service": "Booking",
+      "BookingId": "683EEB2D-B2F3-4075-96EE-788E6E2EED45"
+    }
+    ```
+
+### Single metric with a different dimension
+
+CloudWatch EMF uses the same dimensions across all your metrics. Use **`PushSingleMetric`** if you have a metric that should have different dimensions.
+
+!!! info
+    Generally, this would be an edge case since you [pay for unique metric](https://aws.amazon.com/cloudwatch/pricing). Keep the following formula in mind:
+
+    **unique metric = (metric_name + dimension_name + dimension_value)**
+
+=== "Function.cs"
+
+    ```csharp hl_lines="8-17"
+    using AWS.Lambda.Powertools.Metrics;
+    
+    public class Function {
+      
+      [Metrics(Namespace = ExampleApplication, Service = "Booking")]
+      public async Task<APIGatewayProxyResponse> FunctionHandler(APIGatewayProxyRequest apigProxyEvent, ILambdaContext context)
+      {
+        Metrics.PushSingleMetric(
+                    metricName: "ColdStart",
+                    value: 1,
+                    unit: MetricUnit.Count,
+                    nameSpace: "ExampleApplication",
+                    service: "Booking",
+                    defaultDimensions: new Dictionary<string, string>
+                    {
+                        {"FunctionContext", "$LATEST"}
+                    });
```

Comment:
Please consider naming it as `dimensions`. Single metric is intended to be a `single shot` when creating metrics, so, doesn't make much sense to persist the dimensions.